src/category_theory/groupoid/free_groupoid.lean

```diff
@@ -0,0 +1,190 @@
+/-
+Copyright (c) 2022 Rémi Bottinelli. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémi Bottinelli
+-/
+import category_theory.category.basic
+import category_theory.functor.basic
+import category_theory.groupoid
+import combinatorics.quiver.basic
+import combinatorics.quiver.connected_component
+import logic.relation
+import tactic.nth_rewrite
+import category_theory.path_category
+import category_theory.quotient
+
+/-!
+# Free groupoid on a quiver
+
+This file defines the free groupoid on a quiver, the lifting of a prefunctor to its unique
+extension as a functor from the free groupoid, and proves uniqueness of this extension.
+
+## Main results
+
+Given the type `V` and a quiver instance on `V`:
+
+- `free_groupoid V`: a type synonym for `V`.
+- `free_groupoid_groupoid`: the `groupoid` instance on `free_groupoid V`.
+- `lift`: the lifting of a prefunctor from `V` to `V'` where `V'` is a groupoid, to a functor.
+  `free_groupoid V ⥤ V'`.
+- `lift_spec` and `lift_unique`: the proofs that, respectively, `lift` indeed is a lifting
+  and is the unique one.
```

Comment:
All these links aren't working now but maybe you don't intend them to link to definitions?
For example you need to write [category_theory.groupoid.free.free_groupoid](https://leanprover-community.github.io/mathlib_docs/category_theory/groupoid/free_groupoid.html#category_theory.groupoid.free.free_groupoid) instead of `free_groupoid` for the first link to work. This name feels quite redundant though. You may want to write `def _root_.category_theory.free_groupoid` to get a more succinct name.

`free_groupoid_groupoid` is actually [category_theory.groupoid.free.free_groupoid.category_theory.groupoid](https://leanprover-community.github.io/mathlib_docs/category_theory/groupoid/free_groupoid.html#category_theory.groupoid.free.free_groupoid.category_theory.groupoid) which is super-redundant; you may consider explicitly naming the instance like I suggested above.

`lift_unique` appears to be called `lift_unique_spec` now, and the quotient, path_category, and symmetrify versions have the inconsistent name `lift_spec_unique`. I think we don't need `spec` in the name; if you search mathlib there are lots of things named `lift_unique` but not `lift_spec_unique`/`lift_unique_spec`. It's best to uniformize them in your follow-up PR.